src/services/asset.services.ts

```diff
@@ -1,8 +1,61 @@
 import { Asset } from '../sequelize/models/asset.models';
+import Redis from '../config/redis';
 
 class AssetService {
-  static async getAllAssest() {
-    return Asset.findAll();
+  static async getAllAssets({ offset, limit }): Promise<Asset[]> {
+    let allAssets = await Redis.fetch('allAssets');
+    if (allAssets) {
+      return allAssets;
+    }
+    allAssets = await Asset.findAll({ offset, limit });
+    await Redis.save('allAssets', allAssets);
+
+    return allAssets;
+  }
+
+  static async getAssetBySerialNo(id): Promise<Asset | null> {
+    let asset = await Redis.fetch(`asset:${id}`);
+    if (asset) {
+      return asset;
+    }
+    asset = await Asset.findOne({ where: { serialNo: id } });
+    await Redis.save(`asset:${id}`, asset);
+
+    return asset;
+  }
+
+  static async addAsset(asset): Promise<Asset> {
+    const assetSequelizeObj = await Asset.create(asset);
```

Comment:
What happens in case the asset is already in the system? What do you think by first checking if an asset with a certain serialNo is already in the system?